modules/internal/ChapelBase.chpl

```diff
@@ -720,6 +720,12 @@ module ChapelBase {
     // stone, we could do parallel init for plain old data (POD) types.
     //
 
+    // for uints we need to check that s > 0, so the `s-1` in the following
+    // loops doesn't overflow.
+    if isUint(s) && s == 0 {
+      return;
```

Comment:
You could always return for size 0 regardless of type, but my thinking was there's no point in wasting the time for ints. 

In particular I was thinking that very few 0-sized arrays will be used in practice, so we might as well skip the early exist check to avoid adding any control flow overhead. Also for 0-sized int loops it'll generate a clean c-style for loop that won't have much overhead if there's no iterations anyways. Something like:

```c
for (i=0; i<=1;i+=1)
```